src/content-helper/class-content-helper-feature.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Content Helper: Base class for all Content Helper features
+ *
+ * @package Parsely
+ * @since   3.9.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Content_Helper;
+
+/**
+ * Base class for all Content Helper features.
+ *
+ * @since 3.9.0
+ */
+abstract class Content_Helper_Feature {
+	/**
+	 * Returns the global Content Helper filter name. The global filter controls
+	 * the enabled/disabled state of all Content Helper features.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return string The filter name.
+	 */
+	final public static function get_global_filter_name(): string {
+		return 'wp_parsely_enable_content_helper';
+	}
+
+	/**
+	 * Returns the feature's filter name. The feature filter controls the
+	 * enabled/disabled state of a particular Content Helper feature.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return string The filter name.
+	 */
+	abstract public static function get_feature_filter_name(): string;
+
+	/**
+	 * Returns the feature's script ID.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return string The script ID.
+	 */
+	abstract public static function get_script_id(): string;
+
+	/**
+	 * Returns the feature's style ID.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return string The style ID.
+	 */
+	abstract public static function get_style_id(): string;
+
+	/**
+	 * Runs the feature's initialization process.
+	 */
+	abstract public function run(): void;
+
+	/**
+	 * Returns whether the feature should be enabled.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return bool Whether the feature should be enabled.
+	 */
+	abstract public function should_be_enabled(): bool;
```

Comment:
I've pushed commit https://github.com/Parsely/wp-parsely/pull/1539/commits/bbc4912ea1442e32d35280d533f307c310ff5ccc which uses a [single function](https://github.com/Parsely/wp-parsely/blob/857f1c684f5d04a94cfdf908899b9dde64ce3d3c/src/content-helper/class-content-helper-feature.php#L79-L103) everywhere (except in the widget for convenience purposes). This is simpler and flexible as it will allow for advanced enablement logic that we might have in the future for newer features.